### ISSUE TITLE
mozilla2john errors with python 3.11.6

### ISSUE THREAD

**Issue by Munto** [openwall/john#5431]
mozilla2john is incompatible with python 3.11.6.
Here is the error I got:
```
Traceback (most recent call last):
  File "/usr/bin/mozilla2john", line 73, in <module>
    process_file(sys.argv[i])
  File "/usr/bin/mozilla2john", line 33, in process_file
    idx = data.find("global-salt")
          ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument should be integer or bytes-like object, not 'str'
```
I had to convert string to bytestring in the following lines to make it works:
https://github.com/openwall/john/blob/bleeding-jumbo/run/mozilla2john.py#L33
https://github.com/openwall/john/blob/bleeding-jumbo/run/mozilla2john.py#L43
https://github.com/openwall/john/blob/bleeding-jumbo/run/mozilla2john.py#L58

I don't know if you need to maintain a compatibility with previous python version so I'm just opening an issue.
I use Arch, btw.

**Comment by solardiz** [comment]
Thank you. Confirmed. So basically, we need this? -

```diff
+++ b/run/mozilla2john.py
@@ -30,7 +30,7 @@ def process_file(filename):
 
     data = open(filename, "rb").read()
 
-    idx = data.find("global-salt")
+    idx = data.find(b"global-salt")
     if idx < 0:
         fail(filename)
         return
@@ -40,7 +40,7 @@ def process_file(filename):
         return
     global_salt = data[idx-20:][0:20]  # is the salt always 20 bytes?
 
-    idx = data.find("password-check")
+    idx = data.find(b"password-check")
     if idx < 0:
         fail(filename)
         return
@@ -55,7 +55,7 @@ def process_file(filename):
     version = 3  # fake (this should be improved)
     nnLen = 1  # fake
     oidLen = 11  # fake
-    oidData = "\x00" * oidLen  # fake
+    oidData = b"\x00" * oidLen  # fake
     sys.stdout.write("%s:$mozilla$*%s*%s*%s*" % (os.path.basename(filename), version, len(entry_salt), nnLen))
     sys.stdout.write(hexlify(entry_salt).decode("ascii"))
     sys.stdout.write("*%s*%s*%s*" % (oidLen, hexlify(oidData).decode("ascii"), len(verifier)))
```

> I don't know if you need to maintain a compatibility with previous python version

Preferably, yes, or if we don't then we replace `python` with `python3` in the shebang line. In this case, looking at other Python scripts we have, it appears that the above change would let the script stay compatible with Python 2 (if it currently is? to  be re-tested) and thus to keep it with just `python` for now. Is this understanding correct, @exploide? (I'm not into Python.)

**Comment by solardiz** [comment]
The script patched as above still works fine with Python 2.6, as tested with `john-samples/Mozilla/02-Firefox-password/key3.db`. @Munto please send us a pull request with these changes. Thank you!

**Comment by solardiz** [comment]
Fixed via #5434.

**Comment by Munto** [comment]
Thank you for your endorsement @solardiz and sorry for the non idiomatic commit message 😬
